ambry-api/src/main/java/com.github.ambry/network/NetworkRequestSend.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.network;
+
+/**
+ * An interface to update send time for a network request
+ */
+public interface NetworkRequestSend {
```

Comment:
Still doesn't explain why we need interfaces for this. If you
1. Change the constructor of `NetworkRequestMetrics` to take in `Histogram` instead of `MetricsHistogram`.
2. Change the function names inside `NetworkRequestMetrics` to `updateQueueTime()` and `updateSendTime()`.

and then used the concrete class `NetworkRequestMetrics` and instantiated it with the right histograms, I don't see why it cannot be used both on the server and client side. Are there any problems with that?